description.md

```diff
@@ -0,0 +1,937 @@
+
+The `takerTraits` argument provides a number of options for the taker to choose from during each fill. These options are stored as bit flags and numbers compacted in a `uint256` number. The bit flags in `takerTraits` are arranged in descending order of significance, with the highest bit first.
+
+| Option name | Bit position | Description |
+| --- | --- | --- |
+| MAKER_AMOUNT_FLAG | 255 bit | If set, the protocol implies that the passed amount is the making amount, and the taking amount will be calculated based on the making amount. Otherwise, the passed amount is the taking amount, and the making amount is calculated based on the taking amount. The amount is calculated with AmountCalculator if getters are not set, or with getters provided with an extension by the maker. |
+| UNWRAP_WETH_FLAG | 254 bit | If set, the WETH will be unwrapped into ETH before sending to the taker's target address. |
+| SKIP_ORDER_PERMIT_FLAG | 253 bit | If set, the order skips the maker's permit application. It can be useful to skip the maker's permit application if the permit was already applied during recursive fill. |
+| USE_PERMIT2_FLAG | 252 bit | If set, the order uses the [Uniswap Permit 2](https://github.com/Uniswap/permit2). |
+| THRESHOLD_AMOUNT | 0-251 bit (uint252) | The maximum amount a taker agrees to give in exchange for a making amount. If the calculated taker amount is less than the threshold, then the transaction will be reverted. A zero (0) threshold skips the check. The evaluated equation is<br/>$$  threshold ≤ amount*{takingAmount \over makingAmount} $$ |
+
+# Cancelling an order
+
+There are several ways to cancel an order:
+
+### Automatically (indirect ways)
+
+- **Cancel by expiration deadline**: The simplest way to cancel an order automatically is to set an expiration deadline using the order's `MakerTraits`. Once the expiration date has passed, it is no longer possible to fill the order. Any fill attempts will be reverted with an `OrderExpired` error.
+- **Cancellation by a condition defined in a predicate**: If the condition defined by the order's predicate starts evaluating to false on each check, the order becomes obsolete and cannot be filled anymore. Any fill attempts will be reverted with a `PredicateIsNotTrue` error.
+
+### Manually (direct ways)
+
+Manual methods require sending a cancel transaction, which requires spending gas.
+
+- **Cancel by hash:** The order can be cancelled by directly calling the `cancelOrder` function and passing the `orderHash` and `makerTraits` of the order.
+
+    > **Note**: Orders are cancelled using different invalidators depending on the maker traits flags `ALLOW_MULTIPLE_FILLS` and `NO_PARTIAL_FILL`. Passing wrong traits may result in the call having no effect, and the order will not be cancelled.
+    > 
+
+    The fill attempts will be reverted with a `BitInvalidatedOrder` error if an order doesn't allow either partial or multiple fills, or with an `InvalidatedOrder` error otherwise.
```

Comment:
This is not very clear what this is referring to

Reply:
fixed